test/rules.jl

```diff
@@ -136,4 +136,32 @@ _second(t) = Base.tuple_type_head(Base.tuple_type_tail(t))
         @test ∂self == NoTangent()
         @test ∂x ≈ (1 - 2im) * Ω̄
     end
+
+
+    @testset "@opt_out" begin
+        first_oa(x, y) = x
+        @scalar_rule(first_oa(x, y), (1, 0))
+        @opt_out ChainRulesCore.rrule(::typeof(first_oa), x::T, y::T) where T<:Float32
+        @opt_out(
+            ChainRulesCore.frule(::Any, ::typeof(first_oa), x::T, y::T) where T<:Float32
```

Comment:
If the a general rule has a config, then all more specific rules need to also have a config.
This goes for `@opt_out` rules, as well as normal rules.
It is one of the more annoying things about how `RuleConfig`s work.
But at least it is consistent between rules wither opted out or not.
(I should make a docs PR to emphasize that)

See https://github.com/JuliaDiff/ChainRules.jl/blob/4ad975826ea0639ad709aeb36cc5051b6bf82eaa/src/rulesets/Base/mapreduce.jl#L87-L113
It doesn't often come-up for normal rules often, for calling back into AD since it is basically only if you have a pidgeon whole optimization for a particular function argument.
But it would probably come up more for `@opt_out`, so I will add a special comment.

Can you suggest a test to add?

Reply:
https://github.com/JuliaDiff/ChainRulesCore.jl/pull/403

Reply:
I have added a docs around this